### PR TITLE
Make datepicker respect date formats #1868 fix

### PR DESCRIPTION
**Pull request by defstat**
@asmecher The past 2 days I was trying to find out if there was a case, or an already existing central mechanism that could port php variables in the FormHandler Javascript object, using its "options" parameter (like OJS does with grids and 'features' variable of the 'options' parameter). I couldn't find any.

So I decided to go with a general approach that includes some code changes in form/textInput.tpl and FormHandler.js.

Tried to add the altField and altFormat as well, as @NateWr suggested.

I have found some .datepicker initialization in other places to (other from FormHandler.js), not sure if I have to see those too. I will look at them. I added a function in PKPString that changes PHP date formats to JQuery Datepicker date formats, and registered it in PKPTemplateManager. Not sure if that is the right approach, or the right class to put that kind of a function.

### CODE REVIEW

File: classes/form/FormBuilderVocabulary.inc.php [hunk 1]
```diff
@@ -434,7 +434,6 @@ function _smartyFBVTextInput($params, &$smarty) {
 				case 'value':
 				case 'uniqId':
 					$smarty->assign('FBV_' . $key, $value); break;
-					break;
```
Comment (defstat) [inline classes/form/FormBuilderVocabulary.inc.php]: @NateWr not sure why the code needs a second break there...

Comment (NateWr) [inline classes/form/FormBuilderVocabulary.inc.php]: My bad, I didn't see the other `break;` on the previous line. :)

Comment (defstat) [inline classes/form/FormBuilderVocabulary.inc.php]: Should I leave it as is now? Or take, we will take care of it at another commit?

Comment (NateWr) [inline classes/form/FormBuilderVocabulary.inc.php]: Yeah as you've done it in the commit is good. :+1:

**Comment by defstat** [comment]
@NateWr I changed the initial comment's recommendations. Not sure if I did it correctly for the ojs part. I made a pull request there [https://github.com/pkp/ojs/pull/1082], for the changes that were not in pkp-lib.

**Comment by NateWr** [comment]
Thanks @defstat! I'm taking a look now.

For future reference, it's a good idea to comment in the issue #1868 whenever you're ready for another code review, and to keep a list of all PRs in there. Since the issue is what we track against milestones, it's easiest to keep that as our central record of activity.

**Comment by NateWr** [comment]
Looks great @defstat! Thanks for working through my confusion about the date formatting. :)

I've just got one more request and then I think it's ready for merge. The announcement dates on the frontend templates aren't yet passed through the smarty date formatting. This is my bad, but can you update the templates so it's all together in one PR?

https://github.com/pkp/pkp-lib/blob/master/templates/frontend/objects/announcement_full.tpl#L19
https://github.com/pkp/pkp-lib/blob/master/templates/frontend/objects/announcement_summary.tpl#L24

Once that's done, you'll need to add a submodule commit to the OJS pull request to get the tests running properly. If you're not sure how to do that, maybe Alec can go over it in the dev call tonight? Or I can walk you through it on slack.

**Pull request closed without merging**